pkg/flash/flash_linux.go

```diff
@@ -149,10 +151,19 @@ func (f *Flash) writeAt(p []byte, off int64) (int, error) {
 		{Tx: op.PRDRES.Bytes(), CSChange: true},
 		{Tx: op.WriteEnable.Bytes(), CSChange: true},
 		// Send the address.
-		{Tx: append(op.PageProgram.Bytes(), f.prepareAddress(off)...)},
-		// Send the data.
-		{Tx: p, CSChange: true},
-		{Tx: op.WriteDisable.Bytes(), CSChange: true},
+		{Tx: append(append(op.PageProgram.Bytes(), f.prepareAddress(off)...), p...)},
+	}); err != nil {
+		return 0, err
+	}
+	time.Sleep(10 * time.Microsecond)
+	if err := f.spi.Transfer([]spidev.Transfer{
+		// The meaning of CSChange is ... odd.
+		// IF CSChange is set true here, then CE# never goes
```

Comment:
I think cschange has a different meaning on the last transaction. Let me look at the kernel source and get back...

Reply:
If you `grep cs_change drivers/spi/spi.c`, you'll see the kernel does some extra stuff when it sees this bit, such as adding additional delays.

Reply:
confusing. But at least this works ...